app/controllers/api_controller/parser.rb

```diff
@@ -299,5 +299,11 @@ def fetch_action_hash(aspec, method_name, action_name)
     def collection_option?(option)
       collection_config.option?(@req.collection, option) if @req.collection
     end
+
+    def assert_id_not_specified(data, type)
```

Comment:
Further, I have noticed that some of the create methods are missing this check.

Even when we don't put this into the manager, we could put it into the `parser.rb` there is already a lot of similar checks. The only reason I did not move it, is that I was not convinced it is worth the effort. 

We are doing multiple recursive descents in the ApiController, and it is not easy to tell whether the new place in `parser.rb` does not leave one hole open without the check.